src/System.Text.Json/src/System/Text/Json/Serialization/Converters/JsonPropertyInfoCharNullable.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Reflection;
+using System.Runtime.InteropServices;
+
+namespace System.Text.Json.Serialization.Converters
+{
+    internal class JsonPropertyInfoCharNullable : JsonPropertyInfo<char?>, IJsonValueConverter<char?>
+    {
+        public JsonPropertyInfoCharNullable(Type classType, Type propertyType, PropertyInfo propertyInfo, JsonSerializerOptions options) :
+            base(classType, propertyType, propertyInfo, options)
+        { }
+
+        public bool TryRead(Type valueType, ref Utf8JsonReader reader, out char? value)
+        {
+            if (reader.TokenType != JsonTokenType.String)
+            {
+                value = default;
+                return false;
+            }
+
+            value = reader.GetString()[0];
+            return true;
+        }
+
+        public void Write(char? value, ref Utf8JsonWriter writer)
+        {
+#if BUILDING_INBOX_LIBRARY
+            char tempChar = value.Value;
```

Comment:
It's ok if this throws if !value.HasValue?

Reply:
Yes thanks. Fixing. Looks like this code was copied from the non-nullable char version of this.

Reply:
Update: nulls are pre-processed above this, so a HasValue check isn't necessary. I'll add an assert\comment.